ompi/mpi/c/sendrecv_replace.c

```diff
@@ -76,6 +77,32 @@ int MPI_Sendrecv_replace(void * buf, int count, MPI_Datatype datatype,
         OMPI_ERRHANDLER_CHECK(rc, comm, rc, FUNC_NAME);
     }
 
+#if OPAL_ENABLE_FT_MPI
+    /*
+     * An early check, so as to return early if we are communicating with
+     * a failed process. This is not absolutely necessary since we will
+     * check for this, and other, error conditions during the completion
+     * call in the PML.
+     */
+    if( OPAL_UNLIKELY(!ompi_comm_iface_p2p_check_proc(comm, dest, &rc)) ) {
+        if( MPI_STATUS_IGNORE != status ) {
+            status->MPI_SOURCE = dest;
+            status->MPI_TAG    = recvtag;
```

Comment:
Why do we update the tag if we fail on the send part ?

Reply:
looks wrong.
Also we set MPI_ERROR in a function that returns the error code, which is also wrong.